OpenSim/Common/Path.h

```diff
@@ -101,6 +101,12 @@ class OSIMCOMMON_API Path {
     /// of getInvalidChars()
     bool isLegalPathElement(const std::string& pathElement) const;
 
+    /// Cleans up a path. This includes removing "." and resolving ".." if
+    /// possible (i.e. it will not remove leading ".." but otherwise will
+    /// remove the previous pathElement from _path. This method also checks
+    /// if a path is invalid due to an absolute path starting with "..".
+    void cleanPath();
```

Comment:
The name cleanPath is a bit ambiguous. Maybe ok while the method is private or protected but in public interface I'd lean to more descriptive name (e.g trim.Something, or resolve or Canonicalize etc.)

Reply:
I chatted with @carmichaelong on this and we settled on the name `trimDotAndDotDotElements()`.